exercise-3.sql

```diff
@@ -0,0 +1,107 @@
+-- 1) Write a SQL query to find the total salary of employees who is in Tokyo excluding whose first name is Nancy
+select sum(salary) from (select e.first_name,e.salary,e.department_id,d.location_id from employees as e inner join departments as d on e.department_id = d.department_id inner join locations as l on d.location_id = l.location_id where l.city = 'Seattle' and lower(e.first_name) != 'nancy');
+
+-- 2) Fetch all details of employees who has salary more than the avg salary by each department.
+select * from employees e inner join departments d on e.department_id = d.department_id where e.salary > (select avg(salary) from employees where department_id = e.department_id);
+
+-- 3) Write a SQL query to find the number of employees and its location whose salary is greater than or equal to 7000 and less than 10000
+select l.location_id,count(*) from employees e inner join departments d on e.department_id = d.department_id inner join locations l on d.location_id = l.location_id group by l.location_id;
+
+
+-- 4) Fetch max salary, min salary and avg salary by job and department. 
+--    Info:  grouped by department id and job id ordered by department id and max salary
+select d.department_name,e.job_id,max(e.salary) as max_salary,min(e.salary) as min_salary,avg(e.salary) as avg_salary from employees e inner join departments d on e.department_id = e.department_id group by e.job_id,d.department_name order by d.department_name,max_salary; 
+
+-- 5) Write a SQL query to find the total salary of employees whose country_id is ‘US’ excluding whose first name is Nancy  
+select l.country_id,sum(e.salary) as total_salary from employees e inner join departments d on e.department_id = d.department_id inner join locations l on d.location_id = l.location_id where l.country_id = 'US' and e.first_name != 'Nancy' group by l.country_id;
+
+
+-- 6) Fetch max salary, min salary and avg salary by job id and department id but only for folks who worked in more than one role(job) in a department.
+select d.department_name,e.job_id,max(e.salary) as max_salary,min(e.salary) as min_salary,avg(e.salary) as avg_salary from employees e inner join departments d on e.department_id = e.department_id inner join job_history j on e.employee_id=j.employee_id group by e.job_id,d.department_name order by d.department_name,max_salary; 
+
```

Comment:
but only for folks who worked in more than one role(job) in a department  - Is this condition checked in your query ?